### PR TITLE
feat: add profiling steps

### PR DESCRIPTION
**Pull request by guilhem-barthes**
## Description

Allow transferring function profiling steps through the orchestrator, allowing the builder to register event without having to have http(s) requests to the backend api server (involving having access to the db to get credentials). This also allows having duration for the function available on every backend.

The drawback of this approach is, how it is relying on the event system already in place, the function profiling are also saved in the DB of the orchestrator.

Fixes FL-1519

## Companion PR

- https://github.com/Substra/substra-backend/pull/886

## Checklist

- [ ] [changelog](../CHANGELOG.md) was updated with notable changes
- [ ] documentation was updated

### CODE REVIEW

**Comment by guilhem-barthes** [comment]
/e2e --tests substra-backend=feat/add-profiling-step-functions

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: changes/1519.added [hunk 1]
```diff
@@ -0,0 +1,3 @@
+A new service called `ProfilingService` in charge of propagating profiling
+  - For now, only register and propagates profiling steps
+  - Introduce a new asset type, ASSET_PROFILING_STEP`
```
Comment (SdgJlbl) [inline changes/1519.added:3]:
```suggestion
  - Introduce a new asset type, `ASSET_PROFILING_STEP`
```

File: lib/asset/profiling.proto [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+package orchestrator;
+
+option go_package = "github.com/substra/orchestrator/lib/asset";
+
+import "google/protobuf/empty.proto";
+
+message ProfilingStep {
+    uint32 duration = 1;
+    string asset_key = 2;
+    string step = 3;
```
Comment (SdgJlbl) [inline lib/asset/profiling.proto:12]: I assume the last two items are Enum, but that there is no way of encoding that in protobuf?

File: server/standalone/migration/000064_add_asset_profiling_step.up.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+INSERT INTO asset_kinds(kind) VALUES ('ASSET_PROFILING_STEP')
```
Comment (SdgJlbl) [inline server/standalone/migration/000064_add_asset_profiling_step.up.sql:1]: do we want to add a table to store the Enum with the profiling steps?

**Review by SdgJlbl — APPROVED** [review]
Thanks for your work. A test in the SDK test suite is failing though 🤔

**Comment by SdgJlbl** [comment]
/e2e --tests substra-backend=feat/add-profiling-step-functions

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by guilhem-barthes** [comment]
/e2e --tests substra-backend=feat/add-profiling-step-functions

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: server/standalone/migration/000064_add_asset_profiling_step.up.sql (hunk 3 above)
Comment (guilhem-barthes) [reply server/standalone/migration/000064_add_asset_profiling_step.up.sql:1]: That is the pattern we used for previous enums, and we didn't implemented the management of postgres enums in the code.

**Comment by guilhem-barthes** [comment]
/e2e --tests substra-backend=feat/add-profiling-step-functions

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by guilhem-barthes** [comment]
/e2e --tests sdk,substrafl,frontend,documentation,mnist substra-backend=feat/add-profiling-step-functions

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Pull request merged**